src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -428,24 +428,48 @@ private string StripFormatSpecifier(string exp, out string formatSpecifier)
             }
 
             // array with static size
-            // TODO: could return '(T(*)[n])(exp)' but requires T
             var m = Regex.Match(trimmed, @"^\[?(\d+)\]?$");
             if (m.Success)
             {
+                string count = m.Groups[1].Value; // (\d+) capture group
+                string expr = exp.Substring(0, lastComma);
+
                 if (_engine.DebuggedProcess.MICommandFactory.Mode == MIMode.Gdb)
-                    return "*" + exp.Substring(0, lastComma) + "@" + trimmed;  // this does not work for lldb
+                {
+                    // return *<expression>@<count> which is only supported in GDB
+                    return $"*{expr}@{count}";
+                }
+                else
+                {
+                    string derefType = GetDereferencedTypeString(expr, threadId, frameLevel);
 
-                return exp.Substring(0, lastComma);
+                    // return '*(T(*)[n])(exp)'
+                    return $"*({derefType}(*)[{count}])({expr})";
+                }
             }
 
-
             // array with dynamic size
             if (Regex.Match(trimmed, @"^\[([a-zA-Z_][a-zA-Z_\d]*)\]$").Success)
                 return exp.Substring(0, lastComma);
 
             return exp;
         }
 
+        private string GetDereferencedTypeString(string expr, int threadId, uint frameLevel)
+        {
+            // TODO: Should we error if the current type is not a pointer type?
+
+            Task<string> eval = Task.Run(async () =>
+            {
```

Comment:
Yuck. Can you use WorkerThread.RunOperation?

Reply:
Maybe a better option: delay this work until 'Eval'. Then you would have a reasonable path for handling errors as well.